lib/web/resources_test.go

```diff
@@ -429,6 +435,301 @@ func TestListResources(t *testing.T) {
 	}
 }
 
+// TestListResourcesPing tests for supported and unsupported server versions
+// returned by ping. Unsupported versions should return an error.
+func TestListResourcesPing(t *testing.T) {
+	m := &mockedResourceAPIGetter{}
+
+	m.mockListResources = func(ctx context.Context, req proto.ListResourcesRequest) (*types.ListResourcesResponse, error) {
+		return &types.ListResourcesResponse{}, nil
+	}
+
+	// Test supported version ping.
+	m.mockPing = func(ctx context.Context) (proto.PingResponse, error) {
+		return proto.PingResponse{ServerVersion: "9.1.0"}, nil
+	}
+	mockHTTPReq, err := http.NewRequest("", "", nil)
+	require.NoError(t, err)
+	_, err = listResources(m, mockHTTPReq, types.KindNode)
+	require.NoError(t, err)
+
+	// Test unsupported v8 ping.
+	m.mockPing = func(ctx context.Context) (proto.PingResponse, error) {
+		return proto.PingResponse{ServerVersion: "8.3.1"}, nil
+	}
+	_, err = listResources(m, mockHTTPReq, types.KindNode)
+	require.True(t, trace.IsNotImplemented(err))
```

Comment:
```suggestion
	require.True(t, trace.IsNotImplemented(err), "listResources returned an unexpected error: %v (want not implemented)", err)
```

so, if it fails, it's not "true is not false" or some nonsense.